ggml/src/ggml.c

```diff
@@ -3393,6 +3395,28 @@ struct ggml_tensor * ggml_get_rows_back(
     return result;
 }
 
+// ggml_set_rows
+
+struct ggml_tensor * ggml_set_rows(
+        struct ggml_context * ctx,
+        struct ggml_tensor  * a,
+        struct ggml_tensor  * b,
+        struct ggml_tensor  * c) {
+    GGML_ASSERT(b->ne[2] == c->ne[1]);
+    GGML_ASSERT(c->ne[3] == 1);
+    GGML_ASSERT(a->type == GGML_TYPE_F16);
+    GGML_ASSERT(b->type == GGML_TYPE_F32);
+    GGML_ASSERT(c->type == GGML_TYPE_I64);
```

Comment:
We might want to allow broadcasting `c` into `b`. It would avoid this `ggml_repeat_4d` here:

https://github.com/ggml-org/llama.cpp/blob/a0c0fb6e58137cf1b0a77d657a6924a942c18291/src/llama-kv-cache-unified.cpp#L795-L799